pkg/generators/extension_test.go

```diff
@@ -424,29 +449,81 @@ func TestValidateMemberExtensions(t *testing.T) {
 		}
 	}
 
+}
+
+func TestValidateMemberExtensionsErrors(t *testing.T) {
+
+	listTypeAtomicExtension := extension{
+		idlTag: "listType",
+		xName:  "x-kubernetes-list-type",
+		values: []string{"atomic"},
+	}
+	listTypeMapExtension := extension{
+		idlTag: "listType",
+		xName:  "x-kubernetes-list-type",
+		values: []string{"map"},
+	}
+	listMapKeysExtension := extension{
+		idlTag: "listMapKey",
+		xName:  "x-kubernetes-map-keys",
+		values: []string{"key1"},
+	}
+	listTypeMultipleErrorExtension := extension{
+		idlTag: "listType",
+		xName:  "x-kubernetes-list-type",
+		values: []string{"atomic", "set"},
+	}
+
+	sliceField := types.Member{
+		Name: "Containers",
+		Type: &types.Type{
+			Kind: types.Slice,
+		},
+	}
+	mapField := types.Member{
+		Name: "Containers",
+		Type: &types.Type{
+			Kind: types.Map,
+		},
+	}
+
 	var failureTests = []struct {
 		extensions []extension
 		member     types.Member
 	}{
-		// Test single member extension
+		// List type extension must be defined on a slice.
 		{
-			extensions: []extension{patchStrategyExtension},
+			extensions: []extension{listTypeAtomicExtension},
 			member:     mapField,
 		},
-		// Test multiple member extensions
+		// Map keys defined, but missing listType=map
```

Comment:
There are probably many other use-cases here that are missing. You could almost create a more thorough list, and then use TDD, and maybe comment out those that are not implemented yet.